### PR TITLE
improved util::CpuTimer and added til::printTime

### PR DESCRIPTION
**Pull request by kmuseth**
1) Added util::printTime that outputs nicely-formatted time information. 

2) The API of util::CpuTimer has been improved and is now making use
     a new util::printTime function for nicer output of time information.

### CODE REVIEW

File: openvdb/util/CpuTimer.h [hunk 1]
```diff
@@ -56,65 +57,130 @@ namespace util {
...
-        std::cerr << ostr.str();
+        const double msec = this->milliseconds();
+        printTime(mOutStream, msec, " completed in ", "\n", true, 4, 1);
+        return msec;
     }
 
-private:
+    /// @brief Returns and prints time in milliseconds since construction or start was called.
+    ///
+    /// @note Combine this method with start() to delay output to will completion of task being timed.
```
Comment (danrbailey) [inline openvdb/util/CpuTimer.h]: typo (though I'm not sure what this is intended to be) - "delay output to will completion of task being timed"

File: openvdb/util/Formats.cc [hunk 2]
```diff
@@ -112,10 +112,53 @@ printNumber(std::ostream& os, uint64_t number,
...
+    int group = 0;
+
+    // Write to a string stream so that I/O manipulators like
+    // std::setprecision() don't alter the output stream.
+    std::ostringstream ostr;
+    ostr << head;
+    ostr << std::setprecision(precision) << std::setiosflags(std::ios::fixed);
+
+    double msec = milliseconds;
+    if (msec >= 1000.0) {// avoid unnecessary coverheads
```
Comment (danrbailey) [inline openvdb/util/Formats.cc]: coverheads -> overheads

File: openvdb/util/CpuTimer.h [hunk 3]
```diff
@@ -56,65 +57,130 @@ namespace util {
...
     {
-        this->stop();
-        this->start(msg);
+        tbb::tick_count::interval_t dt = tbb::tick_count::now() - mT0;
+        return 1000.0*dt.seconds();
     }
 
-    /// Return Time diference in milliseconds since construction or start was called.
-    inline double delta() const
+    /// @brief This method is identical to milliseconds() - will soon be deprecated
```
Comment (danrbailey) [inline openvdb/util/CpuTimer.h]: Could we perhaps mark it deprecated now?

File: openvdb/util/CpuTimer.h [hunk 4]
```diff
@@ -56,65 +57,130 @@ namespace util {
...
-    /// @note Should normally be followed by a call to stop()
-    inline void restart(const std::string& msg)
+    /// @note Combine this method with start() to get timing without any outputs.
+    inline double milliseconds() const
     {
-        this->stop();
-        this->start(msg);
+        tbb::tick_count::interval_t dt = tbb::tick_count::now() - mT0;
+        return 1000.0*dt.seconds();
     }
```
Comment (Idclip) [inline openvdb/util/CpuTimer.h]: Any chance we could add a seconds() method to this class too? I'm using this in a few places where the time scale is much larger so end up having to multiple delta() by 1000 everywhere. Could go even further and introduce something like an enum which defines the return scale:

```
inline double detla(const Precision p = MILLISECONDES) {
  tbb::tick_count::interval_t dt = tbb::tick_count::now() - mT0;
  double s = dt.seconds();
  switch (p) {
    case MILLISECONDES : return s*1000.0;
    // etc
   }
}
```

But if not, something which just returns dt.seconds() would be great

File: openvdb/util/Formats.cc [hunk 5]
```diff
@@ -112,10 +112,53 @@ printNumber(std::ostream& os, uint64_t number,
...
+        group = 3;
+      } else if (minutes) {
+        ostr << minutes << " minutes, " << seconds << " seconds, and ";
+        group = 2;
+      } else if (seconds) {
+        ostr << seconds << " seconds and ";
+        group = 1;
+      }
+    }
+    ostr << std::setw(width) << msec << " milliseconds";
```
[the remark this answers is not in the document]
Comment (pcucka) [reply openvdb/util/Formats.cc]: I suppose there are situations where output as verbose as "1 hour, 10 minutes, and 20.5 seconds (4220500.0)" (48 characters in all) is welcome, but I would prefer to at least have the option of the much more compact "1h 10m 20.5s".

[the remark this answers is not in the document]
Comment (kmuseth) [reply openvdb/util/Formats.cc]: adding an option for both a long and compact output format

File: openvdb/util/Formats.cc [hunk 6]
```diff
@@ -112,10 +112,53 @@ printNumber(std::ostream& os, uint64_t number,
...
+      const uint32_t days    = static_cast<uint32_t>(msec / (1000.0*60*60*24));
+      msec -= (seconds + (minutes + (hours + days * 24) * 60) * 60) * 1000.0;
+      if (days) {
+        ostr << days << " days, " << hours << " hours, " << minutes << " minutes, " << seconds << " seconds, and ";
+        group = 4;
+      } else if (hours) {
+        ostr << hours << " hours, " << minutes << " minutes, " << seconds << " seconds, and ";
+        group = 3;
+      } else if (minutes) {
+        ostr << minutes << " minutes, " << seconds << " seconds, and ";
```
[the remark this answers is not in the document]
Comment (kmuseth) [reply openvdb/util/Formats.cc]: fixed

File: openvdb/util/CpuTimer.h (hunk 3 above)
Comment (kmuseth) [reply openvdb/util/CpuTimer.h]: will do

File: openvdb/util/CpuTimer.h (hunk 4 above)
Comment (kmuseth) [reply openvdb/util/CpuTimer.h]: added

Comment (kmuseth) [reply openvdb/util/CpuTimer.h]: I'm deprecating the delta() method in favor of a seconds() and miliseconds() method

**Pull request closed without merging**